start rejecting csrf failures

sravfeyn opened the pull request:
In spite of enabling CSRF on India, no one filed a bug indicating an unexpected behaviour (there were around 130 request rejections). I have analyzed logs from last 3 months [here](https://docs.google.com/spreadsheets/d/1jq1MwUiMmMr0klz5_kh-H54HVbcG2_M1vnPUHlxLBdU/edit) (from india+prod). There are couple of doubtful csrf rejections, these are not reproducible and they even look like expected behaviours. At this point, we should turn on CSRF rejection on Prod for a day/week, let users report a bug if any rejection is unexpected and fix them

Before merging this in, it would be good for anyone to review the [logs analysis](https://docs.google.com/spreadsheets/d/1jq1MwUiMmMr0klz5_kh-H54HVbcG2_M1vnPUHlxLBdU/edit) Best way to review and discuss might be either through doc comments or a quick voice chat.
@czue 
cc @benrudolph

czue commented:
@sravfeyn on my backlog to look through the logs. can definitely commit to getting to it by monday

sravfeyn commented:
(sounds good)

czue commented:
made some comments on the logs doc, but overall looks like a solid analysis and I don't see any major blockers to getting this in.

sravfeyn commented:
@czue I removed the `don't merge` tag since we addressed all issues outlined in the logs analysis docs. This is good to go if you don't see any other blockers

czue commented:
thanks for the great followthrough on this!
